Signal/src/ViewControllers/ConversationView/ConversationViewModel.m

```diff
@@ -654,13 +654,13 @@ - (void)updateViewWithOldItemIdList:(NSArray<NSString *> *)oldItemIdList
     NSSet<NSString *> *newItemIdSet = [NSSet setWithArray:newItemIdList];
 
     // We use sets and dictionaries here to ensure perf.
-    NSMutableSet<NSString *> *deletedItemIdSet = [oldItemIdSet mutableCopy];
+    // We use NSMutableOrderedSet to preserve item ordering.
+    NSMutableOrderedSet<NSString *> *deletedItemIdSet = [NSMutableOrderedSet orderedSetWithArray:oldItemIdList];
     [deletedItemIdSet minusSet:newItemIdSet];
-    NSMutableSet<NSString *> *insertedItemIdSet = [newItemIdSet mutableCopy];
+    NSMutableOrderedSet<NSString *> *insertedItemIdSet = [NSMutableOrderedSet orderedSetWithArray:newItemIdList];
     [insertedItemIdSet minusSet:oldItemIdSet];
-    NSArray<NSString *> *deletedItemIdList = [deletedItemIdSet.allObjects sortedArrayUsingSelector:@selector(compare:)];
-    NSArray<NSString *> *insertedItemIdList =
-        [insertedItemIdSet.allObjects sortedArrayUsingSelector:@selector(compare:)];
```

Comment:
I forgot that we can't safely `compare:` to restore the ordering - these are `NSString`.